include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -185,6 +185,31 @@ def XorROp : UnaryI1ReductionRTLOp<"xorr">;
 // Other Operations
 //===----------------------------------------------------------------------===//
 
+// Extract a range of bits from the specified input.
+def StructExtractOp : RTLOp<"struct_extract", [NoSideEffect]> {
+  let summary = "Extract a named field from a struct.";
+
+  let arguments = (ins StructType:$input, StrAttr:$field);
+  let results = (outs RTLValueType:$result);
+
+  let parser = [{ return ::parse$cppClass(parser, result); }];
+  let printer = [{ return ::print(p, *this); }];
+}
+
+// Extract a range of bits from the specified input.
+def StructInjectOp : RTLOp<"struct_inject", [NoSideEffect, 
+                           AllTypesMatch<["input", "result"]>]> {
+  let summary = "Inject a value into a named field of a struct.";
+
+  let arguments = (ins StructType:$input, StrAttr:$field,
+                   RTLValueType:$newValue);
+  let results = (outs StructType:$result);
+
+  let parser = [{ return ::parse$cppClass(parser, result); }];
+  let printer = [{ return ::print(p, *this); }];
+}
```

Comment:
In addition to the value semantic representation you're going for here, we will eventually need inout semantics operations, similar to the ArrayIndexOp, it would be a StructIndexOp sort of thing, that takes an inout struct and a field name and produces an inout reference to the field.  Definitely fine to be a follow-on patch of course!

Reply:
Follow on.  I'd like to have type constraints of variadic argument nodes first so I don't have to keep having custom parsers and printers.

Reply:
Agreed.